src/System.Net.Security/src/System/Net/SecureChannel.cs

```diff
@@ -1059,16 +1060,12 @@ internal SecurityStatusPal Encrypt(byte[] buffer, int offset, int size, ref byte
                 resultSize = 0;
 
                 int bufferSizeNeeded = checked(size + _headerSize + _trailerSize);
-                if (output != null && bufferSizeNeeded <= output.Length)
-                {
-                    writeBuffer = output;
-                }
-                else
+
```

Comment:
+1 on @stephentoub 's comment. There is nothing common about the way the code is written. It's basically a if (Windows) {} else { //Do *NIX stuff }.